ada/ast.py

```diff
@@ -7272,11 +7273,13 @@ def env_names():
             Self.default_initial_env
         ),
 
-        add_to_env_by_name(
+        add_to_env_kv(
             key=Entity.name_symbol,
             val=Self,
-            name=Self.child_decl_initial_env_name(False),
-            fallback_env=Self.children_env
+            dest_env=Self.child_decl_initial_env_name(False).then(
+                lambda name: named_dest_env(name),
+                default_val=current_dest_env()
+            ),
```

Comment:
It seems that this is now the same expression over and over: what about factoring this into a property that returns a mapping (`T.env_assoc`), and just have `add_to_env(Self.<this-property>)` everywhere?

(or two properties, as we seem to have another series of duplicates at the end of the patch)

Reply:
Good catch, done!